app/backend/src/tests/test_search.py

```diff
@@ -158,6 +160,45 @@ def test_user_filter_meetup_status(db):
         assert [result.user.user_id for result in res.results] == [user_does_not_want_to_meet.id]
 
 
+def test_user_filter_language(db):
+    """
+    Make sure the language filter returns the rigth profiles
+    """
+    recreate_database()
+    user_with_german_fluent, token11 = generate_user()
+    user_with_japanese_conversational, token12 = generate_user()
+
+    with session_scope() as session:
+
+        session.add(
+            LanguageAbility(user_id=user_with_german_fluent.id, language_code="deu", fluency=LanguageFluency.fluent)
+        )
+        session.add(
+            LanguageAbility(
+                user_id=user_with_japanese_conversational.id, language_code="jpn", fluency=LanguageFluency.conversational
+            )
+        )
+
+    with search_session(token11) as api:
+        search_request_1 = search_pb2.UserSearchReq(
+            language_ability_filter=[
+                api_pb2.LanguageAbility(code="deu", fluency=api_pb2.LanguageAbility.Fluency.FLUENCY_FLUENT)
+            ]
+        )
+    with search_session(token12) as api:
+        search_request_2 = search_pb2.UserSearchReq(
+            language_ability_filter=[
+                api_pb2.LanguageAbility(code="jpn", fluency=api_pb2.LanguageAbility.Fluency.FLUENCY_CONVERSATIONAL)
+            ]
+        )
+
+        result_1 = api.UserSearch(search_request_1)
+        result_2 = api.UserSearch(search_request_2)
+
+        assert [result.user.user_id for result in result_1.results] == [user_with_german_fluent.id]
+        assert [result.user.user_id for result in result_2.results] == [user_with_japanese_conversational.id]
+
```

Comment:
If you'd like, it'd be good to have a test where the user is, say `beginner` in a language but you search for `conversational` and they wouldn't show up.